ssl/t1_lib.c

```diff
@@ -692,7 +697,8 @@ int ssl_load_sigalgs(SSL_CTX *ctx)
         if (ctx->ssl_cert_info == NULL)
             return 0;
         for(i = 0; i < ctx->sigalg_list_len; i++) {
-            ctx->ssl_cert_info[i].nid = OBJ_txt2nid(ctx->sigalg_list[i].sigalg_name);
+            const char *keytype = inferred_keytype(&ctx->sigalg_list[i]);
+            ctx->ssl_cert_info[i].pkey_nid = OBJ_txt2nid(keytype);
```

Comment:
Actually, we perhaps have an issue here, what happens if there are multiple signature algorithms for the same underlying key type? I think we should add the key type just once... And perhaps not add it at all if it is one of the standard key types.

Reply:
I think you're confusing things.  SSL_CERT_LOOKUP has one job, to provide access to a public key via a cert, index by key type (NID).  Period.

If multiple signature algorithms are using the same key type (for example, all the sha{nnn}WithRSAEncryption signature algorithms), that should be perfectly fine, they'll simply use the same SSL_CERT_LOOKUP table entry.

I'm not aware if the use of SSL_CERT_LOOKUP array supports multiple certs with the same key type or not.  Perhaps this is supported, but that's out of scope for this particular fix.

Reply:
Yeah well, that's the available input, especially considering the confusion that this PR corrects.

This can probably be refined. Not sure I'm gonna consider it for this PR, but it's certainly an idea for future efforts, if it's worth it.